### DISCUSSION TITLE
AIO Config via Docker ENV VARS?

### DISCUSSION THREAD

**Discussion by Just-Insane** [nextcloud/all-in-one#920] [Ideas] [answered]
Is it possible to configure the AIO settings automatically during install, such as the time zone and selected services (clamav, collabora, etc)?

The use case is that I am trying to install automatically via a script, but it cannot complete automatically without setting up AIO manually in the browser first.

The current workflow is: run install script > open browser and login to AIO > select services/set time zone > start containers > stop containers > modify nextcloud config.php > start containers, which is very clunky.

**Comment by szaimen** [comment]
This is currently not possible.

And I disagree: especially in comparison to other Nextcloud appliances it the current installation flow not clunky at all!

**Reply by DaleJP** [reply]
I have tried a myriad of ways to get Nextcloud running and this is BY FAR the easiest and smoothest if you want HPB for Talk and Files.

**Reply by Just-Insane** [reply]
I’d disagree… it’s clunky if you want to automate the install at all or do anything outside of the “standard”.

**Reply by DaleJP** [reply]
I agree It is a shame that there isn't much room for tweaking and customisation but I think that is the nature of this kind of install. The more complication in the installation, the more room for instability.

The backups from Borg always put it back exactly as I set it up so maybe there is a road to customisation for you there? Just set up a blank template how you want it and put it aside as part of your installation process.

**Reply by dantefromhell** [reply]
Yes please provide a way to fully automate the installation. It will allow people that use automation systems like Ansible to use this docker container much easily.

**Comment by szaimen** [comment, marked as answer]
I will not work on this but PRs are welcome!
